userspace/libscap/engine/gvisor/CMakeLists.txt

```diff
@@ -76,6 +76,14 @@ if (BUILD_SHARED_LIBS)
     add_dependencies(scap_engine_gvisor_o uthash)
     add_dependencies(scap scap_engine_gvisor_o)
     target_sources(scap PRIVATE $<TARGET_OBJECTS:scap_engine_gvisor_o>)
+
+    target_include_directories(scap_engine_gvisor_o
+        PRIVATE
+        ${CMAKE_BINARY_DIR}
+        ${CMAKE_CURRENT_BINARY_DIR}
+        ${CMAKE_SOURCE_DIR}
+        ${CMAKE_SOURCE_DIR}/userspace
+    )
```

Comment:
I'm not sure I'd use `target_include_directories` for bringing in the needed include paths. A `target_link_libraries` line would express the dependency in a better way. 
Is it because of the circular dependency mentioned above?
I'm not objecting to this if there's such a limitation, but let's consider this as a hack to be improved up in the future, perhaps solving the circular dependency (I know it's a harder problem, most likely).

Reply:
I tried that but it didn't have any effect.   Reading on it, it was because an OBJECT "library" is just an object, and is not linked.  So any link directives added to it are ignored by CMake, as far as I understand.